ibm_mq/datadog_checks/ibm_mq/ibm_mq.py

```diff
@@ -192,31 +223,50 @@ def get_pcf_channel_metrics(self, queue_manager, tags, config):
             self.gauge(mname, channels, tags=tags)
 
         # grab all the discoverable channels
-        self._get_channel_status(queue_manager, '*', tags, config)
+        self._submit_channel_status(queue_manager, '*', tags, config)
 
         # check specific channels as well
         # if a channel is not listed in the above one, a user may want to check it specifically,
         # in this case it'll fail
         for channel in config.channels:
-            self._get_channel_status(queue_manager, channel, tags, config)
+            self._submit_channel_status(queue_manager, channel, tags, config)
 
-    def _get_channel_status(self, queue_manager, channel, tags, config):
-        channel_tags = tags + ["channel:{}".format(channel)]
+    def _submit_channel_status(self, queue_manager, search_channel_name, tags, config):
+        """Submit channel status
+        :param search_channel_name might contain wildcard characters
+        """
+        search_channel_tags = tags + ["channel:{}".format(search_channel_name)]
         try:
-            args = {pymqi.CMQCFC.MQCACH_CHANNEL_NAME: ensure_bytes(channel)}
+            args = {pymqi.CMQCFC.MQCACH_CHANNEL_NAME: ensure_bytes(search_channel_name)}
             pcf = pymqi.PCFExecute(queue_manager)
             response = pcf.MQCMD_INQUIRE_CHANNEL_STATUS(args)
+            self.service_check(self.CHANNEL_SERVICE_CHECK, AgentCheck.OK, search_channel_tags)
         except pymqi.MQMIError as e:
             self.log.warning("Error getting CHANNEL stats {}".format(e))
-            self.service_check(self.CHANNEL_SERVICE_CHECK, AgentCheck.CRITICAL, channel_tags)
+            self.service_check(self.CHANNEL_SERVICE_CHECK, AgentCheck.CRITICAL, search_channel_tags)
         else:
             for channel_info in response:
-                name = channel_info[pymqi.CMQCFC.MQCACH_CHANNEL_NAME]
-                name = name.strip()
-
-                # running = 3, stopped = 4
-                status = channel_info[pymqi.CMQCFC.MQIACH_CHANNEL_STATUS]
-                if status == 3:
-                    self.service_check(self.CHANNEL_SERVICE_CHECK, AgentCheck.OK, channel_tags)
-                elif status == 4:
-                    self.service_check(self.CHANNEL_SERVICE_CHECK, AgentCheck.WARNING, channel_tags)
+                channel_name = channel_info[pymqi.CMQCFC.MQCACH_CHANNEL_NAME].decode().strip()
```

Comment:
Nice for the utf-8 `decode`, unicode everywhere 👍 
Quick question, are we sure that the pymqi library only returns bytes here ? If it returns a string, this would fail in py3.

Reply:
Good question. Yes seems so. This part is covered by integration tests, and we have tests are run for both python 2 and 3. So I think it's working fine. 

But better safe than sorry. I update the code to use the utility method called encore_unicode() in few places.